examples/stm32f072_discovery/stm32f072_discovery.hpp

```diff
@@ -0,0 +1,141 @@
+// coding: utf-8
+/* Copyright (c) 2013, Roboterclub Aachen e.V.
+* All Rights Reserved.
+*
+* The file is part of the xpcc library and is released under the 3-clause BSD
+* license. See the file `LICENSE` for the full license governing this code.
+*/
+// ----------------------------------------------------------------------------
+
+//
+// 32F072DISCOVERY
+// Discovery kit for STM32F072 series
+// http://www.st.com/web/en/catalog/tools/FM116/SC959/SS1532/PF259724
+//
+
+#ifndef XPCC_STM32_F072_DISCOVERY_HPP
+#define XPCC_STM32_F072_DISCOVERY_HPP
+
+#include <xpcc/architecture/platform.hpp>
+#include <xpcc/driver/inertial/l3gd20.hpp>
+
+using namespace xpcc::stm32;
+
+
+namespace Board
+{
+
+/// STM32F072 running at 48MHz generated from the internal 48MHz clock
+// TODO: enable once clock driver is implemented
+//using DefaultSystemClock = SystemClock<InternalClock<MHz48>, MHz48>;
+
+template<int F = 8 * 1000 * 1000>
+struct DummyClock {
+	static constexpr int Frequency = F;
+	static constexpr int Usart1 = Frequency;
+	static constexpr int Can1 = Frequency;
+	static constexpr int Spi2 = Frequency;
+};
+
+using DefaultSystemClock = DummyClock<48 * 1000 * 1000>;
+
+using LedUp    = GpioOutputC6;
+using LedDown  = GpioOutputC7;
+using LedLeft  = GpioOutputC8;
+using LedRight = GpioOutputC9;
+using Button   = GpioInputA0;
+
+
+namespace l3g
+{
+using Int1 = GpioInputC1;	// MEMS_INT1 [L3GD20_INT1]: GPXTI0
+using Int2 = GpioInputC2;	// MEMS_INT2 [L3GD20_DRDY/INT2]: GPXTI1
+
+using Cs   = GpioOutputC0;		// CS_I2C/SPI [L3GD20_CS_I2C/SPI]
+using Sck  = GpioOutputB13;	// SPI2_SCK [L3GD20_SCL/SPC]
+using Mosi = GpioOutputB15;	// SPI2_MISO [L3GD20_SDA/SDI/SDO]
+using Miso = GpioInputB14;	// SPI2_MISO [L3GD20_SA0/SDO]
+
+using SpiMaster = SpiMaster2;
+using Transport = xpcc::Lis3TransportSpi< SpiMaster, Cs >;
+using Gyroscope = xpcc::L3gd20< Transport >;
+}
+
+
+inline bool
+enableHSI48()
+{
+	RCC->CR2 |= RCC_CR2_HSI48ON;
+	uint32_t t = 1500;
+	while (!(RCC->CR2 & RCC_CR2_HSI48RDY) and --t) {
+	}
+	return static_cast<bool>(RCC->CR2 & RCC_CR2_HSI48RDY);
+}
+
+inline bool
+switchSystemClockToHSI48()
+{
+	// use HSI48, all prescalers set to 1
+	RCC->CFGR = RCC_CFGR_SW_HSI48;
+	uint32_t t = 1500;
+	// Wait till the main PLL is used as system clock source
+	while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSI48 and --t) {
+	}
+
+	auto success = static_cast<bool>((RCC->CFGR & RCC_CFGR_SWS) == RCC_CFGR_SWS_HSI48);
```

Comment:
This is actually to reduce redundancy, as the type of the variable is unmistakably stated on the right side.
However, this should actually be `const auto`, I will correct that.